### PR TITLE
Smoothen the CLI's interface a bit

### PR DESCRIPTION
**Pull request by thomaseizinger**
- Add `--change-address` to the `buy-xmr` command.
- Remove `--receive-address` from the `resume` command.

### CODE REVIEW

File: swap/src/bin/swap.rs [hunk 1]
```diff
@@ -165,10 +166,6 @@ async fn main() -> Result<()> {
             let seed = Seed::from_file_or_generate(data_dir.as_path())
                 .context("Failed to read in seed file")?;
 
-            if monero_receive_address.network != env_config.monero_network {
```
Comment (da-kami) [inline swap/src/bin/swap.rs:166]: 👍 already done in `cli::command`

**Review by da-kami — APPROVED** [review]

File: swap/src/bitcoin/wallet.rs [hunk 2]
```diff
@@ -309,11 +309,18 @@ where
...
         amount: Amount,
+        change_override: Option<Address>,
     ) -> Result<PartiallySignedTransaction> {
         if self.network != address.network {
             bail!("Cannot build PSBT because network of given address is {} but wallet is on network {}", address.network, self.network);
         }
 
+        if let Some(change) = change_override.as_ref() {
+            if self.network != change.network {
+                bail!("Cannot build PSBT because network of given address is {} but wallet is on network {}", change.network, self.network);
```
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply swap/src/bitcoin/wallet.rs:320]: If I qualify this one, I also have to qualify the above one ...
I think it should be fine. In either case, the user is going to check the networks of the address they supplied.

File: swap/src/cli/command.rs [hunk 3]
```diff
@@ -287,6 +290,12 @@ pub enum RawCommand {
         #[structopt(flatten)]
         bitcoin: Bitcoin,
 
+        #[structopt(
+            long = "change-address",
+            help = "The bitcoin address where any form of change or excess funds should be sent to"
```
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply swap/src/cli/command.rs:290]: I am reworking this as part of the docs. Will make it consistent there!

File: swap/src/bitcoin/wallet.rs [hunk 4]
```diff
@@ -340,6 +347,15 @@ where
             _ => bail!("Unexpected transaction layout"),
         }
 
+        if let ([_, change], [_, psbt_output], Some(change_override)) = (
+            &mut psbt.global.unsigned_tx.output.as_mut_slice(),
+            &mut psbt.outputs.as_mut_slice(),
+            change_override,
+        ) {
+            change.script_pubkey = change_override.script_pubkey();
+            psbt_output.bip32_derivation.clear();
```
[the remark this answers is not in the document]
Comment (thomaseizinger) [reply swap/src/bitcoin/wallet.rs:356]: Can you push a follow-up that makes it clear for you? :)

File: swap/src/bitcoin/wallet.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (da-kami) [reply swap/src/bitcoin/wallet.rs:320]: This one is not super important to me, so feel free to ignore my comment.

**Pull request merged**